File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/AddressAttributes.java
package com.emerchantpay.gateway.genesisandroid.api.interfaces;

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder;

public interface AddressAttributes extends BillingAddressAttributes, ShippingAddressAttributes {
    default RequestBuilder getBillingAddress() {
        return buildBillingAddress();
    }

    default RequestBuilder getShippingAddress() {
        return buildShippingAddress();
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/AddressAttributes.kt
package com.emerchantpay.gateway.genesisandroid.api.interfaces

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder

interface AddressAttributes : BillingAddressAttributes, ShippingAddressAttributes {

    fun getBillingAddress(): RequestBuilder {
        return buildBillingAddress()
    }

    fun getShippingAddress(): RequestBuilder {
        return buildShippingAddress()
    }
}